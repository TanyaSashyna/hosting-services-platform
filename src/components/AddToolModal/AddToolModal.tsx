import React, { FC } from "react";
import { Modal, ButtonGroup, Button } from "@project-name-frontend/ui-package";

type AddToolModalProps = {
  domain: string;
  open: boolean;
  loading: boolean;
  onClose: (state: boolean) => void;
  onAddTool: () => void;
};

export const AddToolModal: FC<AddToolModalProps> = ({
  domain,
  open,
  loading,
  onClose,
  onAddTool,
}) => {
  const handleClose = (): void => {
    !loading && onClose(false);
  };

  return (
    <Modal
      header={`Add Tool to ${domain}`}
      hasCloseButton={true}
      open={open}
      onClose={handleClose}
    >
      By adding Tool, you agree to be billed <strong>29 per month</strong>.
      <ButtonGroup bottomOfCard={true} layout="right">
        {!loading && (
          <Button buttonType="secondary" onClick={handleClose}>
            Cancel
          </Button>
        )}
        <Button onClick={onAddTool} loading={loading} disabled={loading}>
          Add
        </Button>
      </ButtonGroup>
    </Modal>
  );
};

AddToolModal.displayName = "AddToolModal";
