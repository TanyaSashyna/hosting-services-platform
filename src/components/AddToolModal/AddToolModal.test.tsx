import React from "react";
import { cleanup, render, fireEvent } from "@testing-library/react";
import { Page } from "@project-name-frontend/ui-package";
import { AddToolModal } from "./AddToolModal";

afterEach(cleanup);

describe("DnsChangesModal", () => {
  const mockFunc = jest.fn();

  it("renders DnsChangesModal and matches snapshot", () => {
    const { container, asFragment } = render(
      <Page>
        <AddToolModal
          open={true}
          onClose={mockFunc}
          onAddTool={mockFunc}
          domain="example.com"
          loading={false}
        />
      </Page>
    );
    expect(container).toBeDefined();
    expect(asFragment()).toMatchSnapshot();
  });

  it("trigger onClose function on Cancel button click", () => {
    const { getByText } = render(
      <Page>
        <AddToolModal
          open={true}
          onClose={mockFunc}
          onAddTool={mockFunc}
          domain="example.com"
          loading={false}
        />
      </Page>
    );
    const cancelBtn = getByText("Cancel");
    fireEvent.click(cancelBtn);

    expect(mockFunc).toHaveBeenCalled();
  });

  it("trigger onAddTool function on Add button click", () => {
    const { getByText } = render(
      <Page>
        <AddToolModal
          open={true}
          onClose={mockFunc}
          onAddTool={mockFunc}
          domain="example.com"
          loading={false}
        />
      </Page>
    );
    const addBtn = getByText("Add");
    fireEvent.click(addBtn);

    expect(mockFunc).toHaveBeenCalled();
  });

  it("doesn't render Cancel button when loading is true", () => {
    const { container, getByText } = render(
      <Page>
        <AddToolModal
          open={true}
          onClose={mockFunc}
          onAddTool={mockFunc}
          domain="example.com"
          loading={true}
        />
      </Page>
    );

    expect(container.querySelector(".loader")).toBeDefined();
    expect(() => getByText("Cancel")).toThrow("Unable to find an element");
  });
});
