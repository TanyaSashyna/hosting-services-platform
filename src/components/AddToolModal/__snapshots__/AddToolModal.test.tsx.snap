// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DnsChangesModal renders DnsChangesModal and matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="page"
  >
    <div
      class="page__inner"
    >
      <div
        class="page__header"
      >
        <div
          class="header__wrapper"
        >
          <div
            class=""
          >
            <div
              class="header header__top"
            >
              <div
                class="header__left"
              />
              <div
                class="header__middle"
              >
                <div
                  class="header__middle__logo"
                >
                  <div>
                    <svg
                      class="logo__icon logo__icon--undefined"
                      data-name="Layer 1"
                      id="Layer_1"
                      viewBox="0 0 79.13 70.87"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <g
                        id="Artboard-60px-Colour"
                      >
                        <path
                          d="M12.82,1.72c6.83,1.71,14.51,5.06,20.5,12,4.5,5.21,6.84,15.46,6.84,21.46,0,.87-.06,2.59-.06,2.59H38.45a101.22,101.22,0,0,1-11.08-1.72A38.35,38.35,0,0,1,6.84,23.19C1.63,16.92,0,6.37,0,1.72V0H1.71A58.77,58.77,0,0,1,12.82,1.72Z"
                          id="Path"
                        />
                        <path
                          d="M68.85,8c-5.5.92-12.57,4.91-17.47,10.61-3.71,4.34-5.32,11.54-5.32,15.79V37.8h3.45c1.85,0,4.2-.84,6.33-1.28A33.11,33.11,0,0,0,73.26,25.91C79,19,79.13,7.09,79.13,7.09A70.84,70.84,0,0,0,68.85,8Z"
                        />
                        <path
                          d="M20.08,69.37C25.3,67.16,30.32,65.3,35.16,60c3.31-3.67,5-9.57,5-14.6V43.7q-5.29.16-10.54.84C24.27,45.62,20.52,47,15,52.42s-6.56,15.33-6.62,17a11.05,11.05,0,0,0-.07,1.5H9.41C10.63,70.87,16.54,70.87,20.08,69.37Z"
                        />
                        <path
                          d="M57.87,61.42c-3.59-1.51-8-3.37-9.44-7.09A31.52,31.52,0,0,1,46.06,43.7a26.16,26.16,0,0,1,6.7.87,23.67,23.67,0,0,1,10,6c4.52,6.71,3.34,10.31,3.34,12A31.14,31.14,0,0,1,57.87,61.42Z"
                        />
                      </g>
                    </svg>
                  </div>
                  <div
                    class="sideMenu__logoTextContainer"
                  >
                    <svg
                      class="logo__text logo__text--blue"
                      data-name="Layer 1"
                      id="Layer_1"
                      viewBox="78 0 318.89 70.87"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <g
                        id="Artboard-60px-Colour"
                      >
                        <path
                          d="M139.37,62.6v-39h4.72A32.57,32.57,0,0,1,146,26.89c.45,1.06,1,.68,1.67.28A15.8,15.8,0,0,1,160,22.46c8.11.4,14.76,4.88,14.76,15.34V62.6h-9.28V39.51c0-6.11-3.5-8.63-7.66-8.63s-9.23,2.77-9.23,10.34V62.6"
                        />
                        <path
                          d="M261,44.89l-28.88,2a11.2,11.2,0,0,0,4.47,6.8,12.3,12.3,0,0,0,8.15,2.06,12.9,12.9,0,0,0,9.68-5.09l5,5.5a15.67,15.67,0,0,1-5.92,5.33,22.88,22.88,0,0,1-19.31.34A18.14,18.14,0,0,1,226.59,55a22.48,22.48,0,0,1-3.28-10.51A22.91,22.91,0,0,1,225,33.65a18.22,18.22,0,0,1,15.87-11.14,17.67,17.67,0,0,1,16.94,8.88A23.8,23.8,0,0,1,261,42.12C261,42.71,261,43.68,261,44.89ZM234.72,33a12.32,12.32,0,0,0-3,7.64l20.73-1.49a12.12,12.12,0,0,0-3.86-7.11,10.09,10.09,0,0,0-13.93,1Z"
                          id="Shape"
                        />
                        <path
                          d="M218.5,44.89l-28.85,2a11.14,11.14,0,0,0,4.46,6.8,12.26,12.26,0,0,0,8.14,2.06,12.87,12.87,0,0,0,9.64-5.09l5,5.5A15.64,15.64,0,0,1,211,61.47a22.83,22.83,0,0,1-19.29.34A18.26,18.26,0,0,1,184.06,55a22.58,22.58,0,0,1-3.27-10.51,22.69,22.69,0,0,1,1.73-10.79,18.15,18.15,0,0,1,15.86-11.14,17.67,17.67,0,0,1,17,8.88,24.06,24.06,0,0,1,3.15,10.74C218.49,42.71,218.5,43.68,218.5,44.89ZM192.2,33.05a12.29,12.29,0,0,0-3,7.63L210,39.19a12.05,12.05,0,0,0-3.81-7.1,10.06,10.06,0,0,0-13.93,1Z"
                        />
                        <polygon
                          points="266.93 7.09 275.2 7.09 275.2 62.6 266.93 62.6 266.93 7.09"
                        />
                        <path
                          d="M290.27,60a17.8,17.8,0,0,1-6.74-7.16,21.81,21.81,0,0,1-2.42-10.39,21.54,21.54,0,0,1,2.42-10.35A17.9,17.9,0,0,1,290.27,25a19.74,19.74,0,0,1,19.46,0,17.9,17.9,0,0,1,6.74,7.12,21.54,21.54,0,0,1,2.42,10.35,21.81,21.81,0,0,1-2.42,10.39A17.8,17.8,0,0,1,309.73,60,19.74,19.74,0,0,1,290.27,60Zm17.33-8.1a14.16,14.16,0,0,0,3-9.41,14.17,14.17,0,0,0-3-9.41,9.62,9.62,0,0,0-7.66-3.58,9.5,9.5,0,0,0-7.57,3.58,14.16,14.16,0,0,0-3,9.41,14.16,14.16,0,0,0,3,9.41A9.5,9.5,0,0,0,300,55.51,9.63,9.63,0,0,0,307.6,51.93Z"
                          id="Combined-Shape"
                        />
                        <polygon
                          points="134.65 62.6 124.26 62.6 115.29 50.52 106.06 62.6 95.67 62.6 110.61 43.46 95.67 23.62 106.06 23.62 115.29 36.47 124.26 23.62 134.65 23.62 119.97 43.46 134.65 62.6"
                        />
                      </g>
                    </svg>
                  </div>
                </div>
              </div>
              <div
                class="header__right"
              />
            </div>
            <div
              class="header header__bottom"
            >
              <div
                class="header__left"
              />
              <div
                class="header__middle"
              >
                <div
                  class="header__title"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="page__subheader"
        />
      </div>
      <div
        class="page__children"
      >
        <div
          class="page__children__wrapper"
        />
      </div>
      <div
        class="page__footer"
      />
    </div>
    <div
      class="toaster toaster--none"
    >
      <div
        class="toast__wrapper toast__wrapper--top"
      >
        <div
          class="toast toast--default"
        >
          <div
            class="toast__decorator"
          />
          <div
            class="toast__container"
          >
            <i
              class="icon toast__icon"
            >
              info
            </i>
            <div
              class="toast__text"
            >
              <div
                class="toast__text__title"
              />
            </div>
            <button
              class="toast__close"
              type="button"
            >
              <i
                class="icon toast__close__icon"
              >
                close
              </i>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="page__overlay"
    >
      <div
        class="modal__container modal__container--open"
      >
        <div
          class="modal"
          role="dialog"
        >
          <div
            aria-hidden="true"
            class="modal__backdrop"
            role="presentation"
          />
          <div
            class="modal__content modal__content--open"
            role="document"
          >
            <div
              class="card"
            >
              <div
                class="card__head"
              >
                <div
                  class="card__head__text"
                >
                  <div
                    class="card__head--wrap"
                  >
                    <div
                      class="card__title"
                    >
                      <div
                        class="modal__header"
                      >
                        Add Tool to example.com
                        <button
                          class="MuiButtonBase-root MuiIconButton-root iconButton iconButton--default MuiIconButton-sizeSmall"
                          tabindex="0"
                          type="button"
                        >
                          <span
                            class="MuiIconButton-label"
                          >
                            <i
                              class="icon"
                            >
                              close
                            </i>
                          </span>
                        </button>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="card__body"
              >
                By adding Tool, you agree to be billed 
                <strong>
                  R29 per month
                </strong>
                . 
                <div
                  class="buttonGroup buttonGroup--layout-center buttonGroup--mobile buttonGroup--bottomOfCard"
                >
                  <button
                    class="button button--size-default button--type-secondary button--text"
                    type="button"
                  >
                    <div
                      class="button__internal"
                    >
                      <div
                        class="button__children"
                      >
                        Cancel
                      </div>
                    </div>
                  </button>
                  <button
                    class="button button--size-default button--type-default button--text"
                    type="button"
                  >
                    <div
                      class="button__internal"
                    >
                      <div
                        class="button__children"
                      >
                        Add
                      </div>
                    </div>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
